Lib/functools.py

```diff
@@ -906,10 +946,13 @@ def wrapper(*args, **kw):
             raise TypeError(f'{funcname} requires at least '
                             '1 positional argument')
 
-        return dispatch(args[0].__class__)(*args, **kw)
+        from inspect import isclass
+        type_arg = type[arg1] if isclass(arg1 := args[0]) else arg1.__class__
```

Comment:
I see that care was taken in `singledispatch` to only import things when necessary, so if anyone knows a way to get around importing `inspect` for the `isclass` check here, please let me know.

Reply:
Have you looked at the source code?  It’s merely `isinstance(arg, type)`

Reply:
Oh I see, thanks! I replaced it.